addons/binding/org.openhab.binding.yamahareceiver/src/main/java/org/openhab/binding/yamahareceiver/internal/protocol/YamahaReceiverCommunication.java

```diff
@@ -208,6 +213,26 @@ public void updateState(YamahaReceiverState state) throws IOException {
         node = getNode(basicStatus, "Input/Input_Sel_Item_Info/Src_Number");
         value = node != null ? node.getTextContent() : "0";
         state.netRadioChannel = Integer.parseInt(value);
+
+        // Net Radio provides additional informaton like station name and currently played song
+        if ("NET RADIO".equals(state.input)) {
+            try {
+                Document docNet = postAndGetXmlResponse(
+                        "<?xml version=\"1.0\" encoding=\"utf-8\"?><YAMAHA_AV cmd=\"GET\">"
+                                + "<NET_RADIO><Play_Info>GetParam</Play_Info></NET_RADIO></YAMAHA_AV>");
+                Node nodeNetInfo = getNode(docNet.getFirstChild(), "NET_RADIO/Play_Info/Meta_Info");
+                node = getNode(nodeNetInfo, "Station");
+                state.netRadioStation = node != null ? node.getTextContent() : "";
+                node = getNode(nodeNetInfo, "Song");
+                state.netRadioSong = node != null ? node.getTextContent() : "";
+            } catch (IOException e) {
+                // Issue a warning if XML request is not supported by the receiver
+                logger.warn("Request to get additional Net Radio information failed: " + e.getMessage());
```

Comment:
The YamahaReceiverCommunication.java class has no logging on purpose. It is really just for sending/receiving network traffic. Just let the exception be handled by the calling class. If you want to add additional information (like your "Request to get additional Net Radio information failed: " string, handle the exception, and generate a new one but don't do any output here.